### PR TITLE
EPMFARMATS-12756

### PR DESCRIPTION
**Pull request by overdd**

### CODE REVIEW

File: po/AbstractElement.js [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+    };
+
+    async getText() {
+        const elementText = await this.elementFinder.getText();
+        logger.debug(`"${this.name}" element text is [${elementText}]`);
+        return elementText;
+    };
+
+    sendKeys(keys) {
+        logger.debug(`Sending "${keys}" to "${this.name}"`);
```
Comment (SiarheiRylach) [inline po/AbstractElement.js:35]: fyi: better to use clear() before sendKeys()

File: utils/delete_log_files.js [hunk 2]
```diff
@@ -0,0 +1,5 @@
+const fs = require('fs-extra');
+const path = require('path');
+
+const combinedLogFilePath = path.resolve(`./logs/combined.log`);
```
Comment (SiarheiRylach) [inline utils/delete_log_files.js:4]: use usual `''` instead of `` here.

Comment (SiarheiRylach) [inline utils/delete_log_files.js:4]: better to store the name of the log globally in one place

File: po/AbstractElement.js [hunk 3]
```diff
@@ -0,0 +1,45 @@
+const { logger } = require('../config/logger_config');
+const selectorType = require('../config/enum');
```
Comment (SiarheiRylach) [inline po/AbstractElement.js:2]: why do you use it here?

File: po/AbstractPage.js [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+class AbstractPage {
+    constructor(name, url) {
+        this.name = name;
+        this.url = url;
+        this.pageElements = [];
+    };
+
+    addElement(name, selectorType, selector) {
+        logger.debug(`Adding a ${name} page element to page object`);
+        this.pageElements.push(new AbstractElement(name, selectorType, selector));
```
Comment (SiarheiRylach) [inline po/AbstractPage.js:13]: just fyi: abstract means that the item will be only inherited and that's all.

File: EPMFARMATS-12756.js [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+
+
+describe('Microsoft Office support page', function () {
+    const homePage = new SupportMicrosoftHomePage();
+    const searchResultsPage = new SupportMicrosoftSearchResults();
+
+    beforeAll(function () {
+        browser.waitForAngularEnabled(true);
+        homePage.openUrl();
+        browser.manage().window().maximize();
```
Comment (SiarheiRylach) [inline EPMFARMATS-12756.js:21]: Please don't use maximize();

File: EPMFARMATS-12756.js [hunk 6]
```diff
@@ -0,0 +1,65 @@
...
+    });
+
+    it('should get Thank You message after submitting a feedback', () => {
+        homePage.getElement('mainSearchBox').click();
```
Comment (SiarheiRylach) [inline EPMFARMATS-12756.js:25]: use async\await to avoid unexpected behavior.

File: EPMFARMATS-12756.js [hunk 7]
```diff
@@ -0,0 +1,65 @@
...
+        browser.manage().window().maximize();
+    });
+
+    it('should redirect to search results page after search data submit', async () => {
+        await homePage.getElement('toAirportName').click();
+        await homePage.getElement('toAirportNameSearchInput').sendKeys('Berlin \n');
+        await browser.wait(EC.visibilityOf(homePage.getElement('berlinCityAirport').elementFinder));
+        await homePage.getElement('berlinCityAirport').getText()
+            .then(text => {
+                text.includes('BER') ? homePage.getElement('berlinCityAirport').click() : console.log('Wrong search result');
```
Comment (SiarheiRylach) [inline EPMFARMATS-12756.js:52]: logger?

File: EPMFARMATS-12756.js [hunk 8]
```diff
@@ -0,0 +1,65 @@
...
+            });
+        await browser.wait(EC.visibilityOf(homePage.getElement('tripTypeSelector').elementFinder));
+        await homePage.getElement('tripTypeSelector').click();
+        await homePage.getElement('tripTypeOneWay').click();
+        await homePage.getElement('departmentDateLabel').click();
+        await homePage.getElement('departureDate').click();
+        await homePage.getElement('departureDatePickerDone').click();
+        await homePage.getElement('buttonBookSubmit').click();
+        await browser.wait(EC.visibilityOf(bookAFlight.getElement('advancedSearchHeading').elementFinder));
+        await console.log("Ready!");
```
Comment (SiarheiRylach) [inline EPMFARMATS-12756.js:62]: why await for console.log?

File: po/AbstractElement.js (hunk 1 above)
Comment (overdd) [reply po/AbstractElement.js:35]: Changed:
    sendKeys(keys) {
        this.elementFinder.clear();
        logger.debug(`Sending "${keys}" to "${this.name}"`);
        return this.elementFinder.sendKeys(keys);
    }

File: utils/delete_log_files.js (hunk 2 above)
Comment (overdd) [reply utils/delete_log_files.js:4]: Done

File: po/AbstractElement.js (hunk 3 above)
Comment (overdd) [reply po/AbstractElement.js:2]: I store enum with types of selectors globally, for easy access

File: utils/delete_log_files.js (hunk 2 above)
Comment (overdd) [reply utils/delete_log_files.js:4]: log path is now stored at config/logger_config.js

File: po/AbstractPage.js (hunk 4 above)
Comment (overdd) [reply po/AbstractPage.js:13]: Got it

File: EPMFARMATS-12756.js (hunk 5 above)
Comment (overdd) [reply EPMFARMATS-12756.js:21]: Fixed

File: EPMFARMATS-12756.js (hunk 6 above)
Comment (overdd) [reply EPMFARMATS-12756.js:25]: fixed

File: EPMFARMATS-12756.js (hunk 8 above)
Comment (overdd) [reply EPMFARMATS-12756.js:62]: Removed

File: EPMFARMATS-12756.js (hunk 7 above)
Comment (overdd) [reply EPMFARMATS-12756.js:52]: Switched to throw Error - test is supposed to fail

**Review by SiarheiRylach — APPROVED** [review]

**Pull request merged**